src/qforte/abc/uccvqeabc.py

```diff
@@ -312,34 +312,29 @@ def report_iteration(self, x):
         self._k_counter += 1
 
         if self._k_counter == 1:
-            print(
-                "\n    k iteration         Energy               dE           Ngvec ev      Ngm ev*         ||g||"
-            )
-            print(
-                "--------------------------------------------------------------------------------------------------"
-            )
+            header = "\n    k iteration         Energy               dE"
+            if self._use_analytic_grad:
```

Comment:
Looking at `uccnvqe.py`, it appears that a user could potentially request `optimizer = "Nelder-Mead"` and `use_analytic_grad = True`. More or less the same applies to `adaptvqe.py`. To ensure accurate printing of diagnostics, I suggest internally setting `self._use_analytic_grad` to `False` if a gradient-free optimizer is selected. Additionally, a warning message could be printed to inform the user of this change.